Support NPM installation switch

mikezhu-dd opened the pull request:
Tested via local change:
default['datadog']['windows_agent_url'] = 'https://s3.amazonaws.com/dd-agent-mstesting/builds/tagged/'
default['datadog']['windows_npm_install'] = true
default['datadog']['system_probe']['enabled'] = true
dd_agent_installer = "datadog-agent-7.27.0-rc.4-1-x86_64.msi"

and running command:
kitchen converge dd-agent-windows-2012r2-13113

Observed the VM provisioned properly, NPM installed and "C:/ProgramData/Datadog/system-probe.yaml" is created.

albertvaka reviewed (changes requested):
LGTM, but I think we can improve the user experience:

NPM only runs if `system-probe.yaml` contains a section like this:
```
network_config:
    enabled: true
``` 

Although it's technically possible to use Windows NPM with the current PR, you need to do three things:
- Setting `windows_npm_install = true` to install the driver
- Setting `system_probe.enabled = true` to create the system-probe.yaml file
- Setting `system_probe.extra_config: {"network_enabled": { "enabled": true } }` to write the above section in the file.

We can simplify this to a great extent with the addition of a new chef property:

```
default['datadog']['system_probe']['network_enabled'] = true
```

This property could have the three effects mentioned: installing the driver, enabling the system-probe.yaml creation and writing the above section to the file. We can maintain the existing properties (eg: `windows_npm_install`) for users who want full control, but for most cases being able to set a single property will simplify the user experience.

mikezhu-dd commented:
Found and fixed some issues:
1. Sysprobe recipe only worked on Linux with agent version > 6.11. This has been now fixed to allow it work on Windows with agent version > 7.26. In the first commit, system_probe.yaml appeared during test, but it was from the MSI.
2. system_probe.yaml.erb loads values directly (except for extra config), so the code in system-probe.rb setting the config file never worked. This has been fixed to load values from ruby file.

Tested via local changes below:
default['datadog']['agent_version'] = '7.27.0'
default['datadog']['windows_agent_url'] = 'https://s3.amazonaws.com/dd-agent-mstesting/builds/tagged/'
default['datadog']['system_probe']['network_enabled'] = true
dd_agent_installer = "datadog-agent-7.27.0-rc.4-1-x86_64.msi"

albertvaka commented on recipes/dd-agent.rb:
```diff
@@ -158,11 +158,12 @@ def template_vars
 
 system_probe_managed = node['datadog']['system_probe']['manage_config']
 agent_version_greater_than_6_11 = agent_major_version > 5 && (agent_minor_version.nil? || agent_minor_version > 11) || agent_major_version > 6
+agent_version_greater_than_7_26 = agent_major_version > 6 && (agent_minor_version.nil? || agent_minor_version > 26) || agent_major_version > 7
```
```suggestion
agent_version_greater_than_7_26 = agent_major_version > 5 && (agent_minor_version.nil? || agent_minor_version > 26) || agent_major_version > 7
```
Both 6.27 and 7.27 support NPM.

mikezhu-dd commented on recipes/system-probe.rb:
```diff
@@ -35,21 +43,34 @@
 
+  if node['datadog']['system_probe']['network_enabled']
+    # if network monitoring need to be enabled, enable it in the config yaml file.
+    extra_config['network_enabled'] = true
```
This is now fixed. Below is the new yaml file generated:
system_probe_config:
  enabled: false
  sysprobe_socket: "/opt/datadog-agent/run/sysprobe.sock"
  debug_port: 0
  bpf_debug: false
  enable_conntrack: false
network_config:
  enabled: true

mikezhu-dd commented on recipes/system-probe.rb:
```diff
@@ -35,21 +43,34 @@
   source 'system_probe.yaml.erb'
   variables(
-    enabled: node['datadog']['system_probe']['enabled'],
+    enabled: sysprobe_enabled,
```
Thanks, now fixed.

albertvaka commented on recipes/dd-agent.rb:
```diff
@@ -158,11 +158,12 @@ def template_vars
 
 system_probe_managed = node['datadog']['system_probe']['manage_config']
 agent_version_greater_than_6_11 = agent_major_version > 5 && (agent_minor_version.nil? || agent_minor_version > 11) || agent_major_version > 6
+agent_version_greater_than_6_26 = agent_major_version > 5 && (agent_minor_version.nil? || agent_minor_version > 26) || agent_major_version > 6
```
```suggestion
agent_version_greater_than_6_26 = agent_major_version > 5 && (agent_minor_version.nil? || agent_minor_version > 26)
```
Removed `> 6` because we don't want to match all `7.x` versions.

albertvaka reviewed (approved):
Left one comment but otherwise LGTM.
